code/datums/gamemodes/game_mode.dm

```diff
@@ -708,6 +710,35 @@ Sensors indicate [numXenosShip ? "[numXenosShip]" : "no"] unknown lifeform signa
 
 	return list(num_humans, num_xenos)
 
+/datum/game_mode/proc/get_total_joblarvaworth(list/z_levels = SSmapping.levels_by_any_trait(list(ZTRAIT_MARINE_MAIN_SHIP, ZTRAIT_GROUND, ZTRAIT_RESERVED)), count_flags)
+	. = 0
+
+	for(var/i in GLOB.alive_human_list)
+		var/mob/living/carbon/human/H = i
+		var/datum/job/job = SSjob.GetJob(H.job)
+		if(count_flags & COUNT_IGNORE_HUMAN_SSD && !H.client)
+			continue
+		if(H.status_flags & XENO_HOST)
+			continue
+		if(!(H.z in z_levels) || isspaceturf(H.loc))
+			continue
+		. += job.larvaworth
+
+/datum/game_mode/proc/balance_scales()
+	var/datum/hive_status/normal/xeno_hive = GLOB.hive_datums[XENO_HIVE_NORMAL]
+	var/num_xenos = xeno_hive.get_total_xeno_number() - length(xeno_hive.get_ssd_xenos()) + xeno_hive.stored_larva
+	latejoin_larvapoints = (get_total_joblarvaworth() - (num_xenos * latejoin_larvapoints_required)) / latejoin_larvapoints_required
+	if(!num_xenos) //Distress ends here
+		if(!length(GLOB.xeno_resin_silos)) //Crash ends here
+			check_finished(TRUE)
+			return //RIP benos.
+		if(xeno_hive.stored_larva)
+			return //No need for respawns nor to end the game. They can use their burrowed larvas.
+		xeno_hive.stored_larva += max(1, round(latejoin_larvapoints)) //At least one
+		return 
+	if(latejoin_larvapoints < 1)
+		return //Things are balanced, no burrowed needed
+	xeno_hive.stored_larva += round(latejoin_larvapoints) //however many burrowed they can afford to buy, floored
 
```

Comment:
`latejoin_larvapoints` are added but are not substracted after the `stored_larva` count increases.
For distress this will call extra larvas.